internal/pkg/cli/file/watch_integration_test.go

```diff
@@ -72,60 +73,71 @@ func TestRecursiveWatcher(t *testing.T) {
 
 	t.Run("Watch", func(t *testing.T) {
 		// SETUP
-		err := watcher.Add(fmt.Sprintf("%s/watch", tmp))
+		err := watcher.Add(filepath.ToSlash(filepath.Join(tmp, "watch")))
 		require.NoError(t, err)
 
 		eventsCh := watcher.Events()
 		errorsCh := watcher.Errors()
 
-		expectAndPopulateEvents := func(t *testing.T, n int, events map[fsnotify.Event]struct{}) {
-			for i := 0; i < n; i++ {
+		// expectNextEvents consumes the watcher's event channel to retrieve the next n events in expectedEvents in no particular order.
```

Comment:
```suggestion
		// expectNextEvents consumes the watcher's event channel to retrieve the next n events in eventsExpected in no particular order.
```
Why do we not care about the order?

Reply:
The `rename` operation causes both a `rename` and `create` event which can be sent in any order. So, this function necessarily cannot care about order.

Reply:
gotcha! in this case `rename` and `create`'s order doesn't matter, but if we have a `os.Chmod` operation followed by a `os.Rename` operation, we still care that `fsnotify.Chmod` goes before `fsnotify.Rename` and `fsnotify.Create` - is that right?

Reply:
> we still care that `fsnotify.Chmod` goes before `fsnotify.Rename` and `fsnotify.Create`

Ideally yes, I tinkered with some implementations but I felt that maintaining this in the test wasn't quite as important as making it so that the test is easier to understand.